src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,401 @@
+                "$1$2(+12)$4$5$6$7$8$9",
+                "$1$2)$3$4$5$6$7$8$9",
+                "$1$2$3($4$5$6$7$8$9",
+                "$1$2)$3($4$5$6$7$8$9",
+                "$1$2$3$4($5)$6$7$8$9",
+                "$1$2$3$4$5($6)$7$8$9",
+                "+ 38$2$3$4$5$6$7$8$9",
+                "++38$2$3$4$5$6$7$8$9",
+                "+3 8$2$3$4$5$6$7$8$9",
+                "$1$2 0 50 $4$5$6$7$8$9",
+                "$1$2 05 0 $4$5$6$7$8$9",
+                "$1$2$3$4 67 8 $6$7$8$9",
+                "$1$2$3$4 6 7 8 $6$7$8$9");
+
+        List<String> tempList = new ArrayList<>();
+        for (String wrongNumber : wrongPhoneNumbers) {
+            addReplacedPhoneNumbers(replacementList, tempList, wrongNumber);
+        }
+
+        List<String> specialChars = getSpecialCharacters("\\");
+        specialChars.removeAll(List.of("\\.", "\\-"));
+
+        specialChars.forEach(
+                specialChar -> wrongPhoneNumbers.forEach(
+                        number -> {
+                            tempList.add(number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                    "$1$2$3$4$5".concat(specialChar).concat("$7$8$9")));
+
+                            String twoDigits = number
+                                    .replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$7");
+                            tempList.add(number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                    "$1$2$3$4$5$6".concat("\\"
+                                            .concat(twoDigits.replaceAll("^\\d",
+                                                    specialChar))).concat("$8$9")));
+                        }));
+
+        wrongPhoneNumbers.clear();
+        wrongPhoneNumbers.addAll(tempList);
+
+        return wrongPhoneNumbers;
+    }
+
+    private static List<String> createCorrectUserNames() {
+        List<String> correctNames = new ArrayList<>(List.of(
+                "pushkin",
+                getRandomString(ENGLISH_LETTERS_AND_NUMBERS, 500)));
+
+        IntStream.range(0, 10).mapToObj(String::valueOf).forEach(correctNames::add);
+        List<String> specialChars = getSpecialCharacters("");
+        specialChars.addAll(getRussianAndUkrLetters());
+
+        correctNames.addAll(specialChars);
+
+        return correctNames;
+    }
+
+    private static List<String> createWrongUserNames() {
+
+        return List.of(
+                "",
+                " ",
+                "www bbb",
+                " www",
+                "www ");
+    }
+
+    private static List<String> createCorrectWordsEmptyOrMin2Max50() {
+        List<String> correctSymbols = List.of(
+                "w",
+                "W",
+                "0",
+                "9",
+                "-",
+                "'",
+                "`",
+                "_"
+        );
+
+        List<String> tempList = new ArrayList<>();
+
+        IntStream.range(0, correctSymbols.size()).forEach(i -> {
```

Comment:
Just use loop with index. You don't use IntStream features here at all. And I already saw similar logic so you should refactor it as well.